aderyn_core/src/detect/detector.rs

```diff
@@ -266,4 +273,91 @@ pub mod detector_test_helpers {
         });
         context
     }
+
+    pub fn load_contract_directly(filepath: &str) -> WorkspaceContext {
+        let solidity_file = &ensure_valid_solidity_file(filepath);
+        let solidity_content = std::fs::read_to_string(solidity_file).unwrap();
+
+        let compiler_input = CompilerInput::new(solidity_file.as_path()).unwrap();
+        let compiler_input = compiler_input.first().unwrap(); // There's only 1 file in the path
+        let version = Solc::detect_version(&Source {
+            content: Arc::new(solidity_content.clone()),
+        })
+        .unwrap();
+
+        let solc = Solc::find_or_install_svm_version(format!("{}", version)).unwrap();
+        let solc_bin = solc.solc.to_str().unwrap();
+
+        let file_arg = compiler_input
+            .sources
+            .first_key_value()
+            .unwrap()
+            .0
+            .to_str()
+            .unwrap();
+
+        let command = Command::new(solc_bin)
+            .args(["--ast-compact-json", file_arg])
+            .current_dir("/")
+            .stdout(Stdio::piped())
+            .output();
+
+        if let Ok(command) = command {
+            let stdout = String::from_utf8(command.stdout).unwrap();
+
+            let mut pick_next_line = false;
+            let mut ast_content = String::new();
+            for line in stdout.lines() {
+                if line.starts_with("======= ") {
+                    let end_marker = line.find(" =======").unwrap();
+                    let filepath = &line["======= ".len()..end_marker];
+                    if file_arg.contains(filepath) {
```

Comment:
This loop seems to exclude base contracts.

In the previous step, solc outputs AST for all inherited and target contracts. This line only picks the one that matches the filename, meaning that `ast_content` doesn't include the base contracts, and we don't get the inherited details of those base contracts.